File: dataset/output/MISTRAL_codestral-latest/Task34_MISTRAL_codestral-latest.cpp
#include <cmath>
#include <string>

bool narcissistic(int value) {
    std::string str_value = std::to_string(value);
    int n = str_value.length();
    int sum = 0;

    for (char c : str_value) {
        sum += std::pow(c - '0', n);
    }

    return value == sum;
}

File: dataset/output/MISTRAL_codestral-latest/Task34_MISTRAL_codestral-latest.java
public class NarcissisticNumber {
    public static boolean isNarcissistic(int number) {
        int originalNumber, result = 0, remainder;
        int n = Integer.toString(number).length();

        originalNumber = number;

        while (originalNumber != 0) {
            remainder = originalNumber % 10;
            result += Math.pow(remainder, n);
            originalNumber /= 10;
        }

        return number == result;
    }
}